### PR TITLE
[Core: Utility, Menu Filter] Move functions from Utility to Menu Filter because they're not used elsewhere

### PR DESCRIPTION
**Pull request by johnsaigle**
### Brief summary of changes

`arrayToCsv` and `getCleanString` are only used by `NDB_Menu_Filter` and so they should go there instead of `Utility`.

This PR is based on #4118 as that PR removes the only occurrence of `getCleanString()` that occurs elsewhere.

### CODE REVIEW

**Comment by driusan** [comment]
Can you rebase this? #4118 is merged.

File: php/libraries/NDB_Menu_Filter.class.inc [hunk 1]
```diff
@@ -734,7 +734,51 @@ class NDB_Menu_Filter extends NDB_Menu
...
+    /**
+    * Coverts array into CSV string
+    *
+    * @param array $array the array to be converted
+    *
+    * @return string CSV string of data
+    */
+    function arrayToCSV(array $array): string
+    {
+        $fp = fopen("php://temp", 'w+');
```
Comment (paiva) [inline php/libraries/NDB_Menu_Filter.class.inc]: Is is necessary to do a try/catch or not necessary because it is on temp?

Comment (johnsaigle) [reply php/libraries/NDB_Menu_Filter.class.inc]: I'm not sure I didn't write this code. This PR is just moving it from one place to another.

Comment (PapillonMcGill) [reply php/libraries/NDB_Menu_Filter.class.inc]: Unless the server run out of memory, there should not be a problem. Even in that case, even a try/catch would not make much difference.

**Pull request merged**